src/interp.rs

```diff
@@ -25,13 +25,69 @@ use std::f64::consts::PI;
 const ALMOST_ZERO: f64 = 0.000001;
 const TAPS: usize = 48;
 
+trait ArrayBuf<Item>: std::borrow::BorrowMut<[Item]> {
```

Comment:
Maybe add a comment here that this can be made nicer with const generics (instead of duplicating the length into an associated constant)